APCAppCore/APCAppCore/Library/APCTaskImporter.m

```diff
@@ -0,0 +1,489 @@
+                                        nestedError: errorLoadingSurveyFile];
+            
+            APCLogError2 (error);
+        }
+        
+        else
+        {
+            NSError *errorParsingSurveyContent = nil;
+            NSDictionary *surveyContent = [NSJSONSerialization JSONObjectWithData: jsonData
+                                                                          options: 0
+                                                                            error: & errorParsingSurveyContent];
+            if (! surveyContent)
+            {
+                NSError *error = [NSError errorWithCode: APCErrorParsingSurveyContentCode
+                                                 domain: APCErrorDomain
+                                          failureReason: APCErrorParsingSurveyContentReason
+                                     recoverySuggestion: APCErrorParsingSurveyContentSuggestion
+                                            nestedError: errorParsingSurveyContent];
+                
+                APCLogError2 (error);
+            }
+            
+            else
+            {
+                @try
+                {
+                    id manager = SBBComponent(SBBSurveyManager);
+                    SBBSurvey *survey = [[manager objectManager] objectFromBridgeJSON: surveyContent];
+                    rkSurvey = [APCTask rkTaskFromSBBSurvey: survey];
+                }
+                @catch (NSException *exception)
+                {
+                    NSError *error = [NSError errorWithCode: APCErrorLoadingNativeBridgeSurveyObjectCode
+                                                     domain: APCErrorDomain
+                                              failureReason: APCErrorLoadingNativeBridgeSurveyObjectReason
+                                         recoverySuggestion: APCErrorLoadingNativeBridgeSurveyObjectSuggestion
+                                            relatedFilePath: surveyFilePath
+                                                 relatedURL: nil
+                                                nestedError: nil
+                                              otherUserInfo: @{ @"exception": exception,
+                                                                @"stackTrace": exception.callStackSymbols }];
+                    
+                    APCLogError2 (error);
+                }
+                @finally
+                {
+                    
+                }
+            }
+        }
+    }
+    
+    return rkSurvey;
+}
+
+// ---------------------------------------------------------
+#pragma mark - JSON conversions
+// ---------------------------------------------------------
+
+/**
+ Convert inbound Bridge server data to an NSDictionary of keys we know how to
+ look for.
+ 
+ This lets us use the same method to process data downloaded from the server as
+ we do data pulled from a local JSON file. It also lets us flatten out the
```

Comment:
My thinking on this was that swapping out how we interact with the Bridge SDK is probably a good idea, but seemed a bit beyond the scope of an already large chunk of work. I'd be totally up for adopting a more direction interaction, but there were enough question marks around that like "How do I do it?", "What about values that don't come from Bridge that we're setting manually?" and "How do I insure that valid values are not overwritten with nil?" that I was hesitant to include it in this body of work.